File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        std::string palindrome;
        
        for(const char ch : s){
            if(std::isalnum(ch)){
                palindrome += std::tolower(ch);
            }
        }
        std::string temp(palindrome);
        std::reverse(palindrome.begin(),palindrome.end());
        return temp == palindrome;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.cs
public class Solution {
    public bool IsPalindrome(string s) {
        int l = 0;
        int r = s.Length - 1;

        while (l <= r) {
            while (!Char.IsLetterOrDigit(s[l]) && l < r) l++;
            while (!Char.IsLetterOrDigit(s[r]) && l < r) r--;
            if (Char.ToLower(s[l]) != Char.ToLower(s[r])) return false;
            l++;
            r--;
        }

        return true;
    }
}